scripts/build.d/gmsh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+pkgname=gmsh
+pkgbranch=${VERSION:-master}
+pkgfull=$pkgname-$pkgbranch
+
+# unpack (clone)
+syncgit https://gitlab.onelab.info/gmsh/ ${pkgname} ${pkgbranch}
+
+pushd ${DEVENVROOT}/flavors/${DEVENVFLAVOR}/src/${pkgfull} > /dev/null
+
+  mkdir -p build
+
+  cd build
+
+  cmakecmd=("cmake")
+  cmakecmd+=("-DCMAKE_INSTALL_PREFIX=${DEVENVPREFIX}")
```

Comment:
It looks like need to add  `cmakecmd+=("-DBUILD_SHARED_LIBS=ON")` to enable shared library?

Reply:
Thanks for the review. This is a question in my mind as well. What is the reasoning to enable shared library by default? Is it expected by the architecture of `devenv`? For example, is it because we want to open API for Python/C so we would like to enable shared library?

Reply:
Dynamic linking provides more flexibility when loading a program, and saves significant memory where there are multiple copies of the program that uses the same library.  I think shared objects should be the default mode unless there are other concerns, e.g., resolving ABI conflicts.

Reply:
Understood. It sounds making sense to me. I will update the pull request and document the rationale in sections of README like "contribution" or "development".